### PR TITLE
Pid refactoring

### PR DESCRIPTION
**Pull request by danielemorra98**
I moved the location of `PIDConfiguration.*` from `VehicleDynamics/DriveModels` to `Utilities/Controllers` so that it could be reused by other components without referring to VehicleDynamics dependency.

### CODE REVIEW

**Comment by adamdbrw** [comment]
@danielemorra98 could you please cast this against development and only keep the actual change in the PR?

File: Gems/ROS2/Code/Source/Lidar/LidarRaycaster.h [hunk 1]
```diff
@@ -12,44 +12,37 @@
 #include <AzCore/Math/Vector3.h>
 #include <AzCore/std/containers/vector.h>
 #include <AzFramework/Physics/PhysicsScene.h>
+#include <ROS2/Lidar/LidarRaycasterBus.h>
```
Comment (michalpelka) [inline Gems/ROS2/Code/Source/Lidar/LidarRaycaster.h:15]: Why this file is touched by this PR?

File: Gems/XR/Code/Include/XR/XRSystem.h [hunk 2]
```diff
@@ -81,6 +81,7 @@ namespace XR
         float GetYJoyStickState(AZ::u32 handIndex) const override;
         float GetSqueezeState(AZ::u32 handIndex) const override;
         float GetTriggerState(AZ::u32 handIndex) const override;
+        AZ::Data::Instance<AZ::RPI::AttachmentImage> InitPassFoveatedAttachment(const AZ::RPI::PassTemplate& passTemplate, const AZ::RHI::XRFoveatedLevel* level = nullptr) const override;
```
Comment (michalpelka) [inline Gems/XR/Code/Include/XR/XRSystem.h:84]: Why this file is touched by the PR at all?

**Review by michalpelka — CHANGES_REQUESTED** [review]
I think PR is containing changes that are added by mistake.
Please adjust the contribution to the intended changes.

**Comment by hultonha** [comment]
@danielemorra98 Did you see @michalpelka's comment? Can any accidentally added files be removed from this PR please? Thanks!

**Pull request closed without merging**